app/src/main/java/com/xzyht/notifyrelay/sync/ConnectionKeepAlive.kt

```
@@ -71,12 +73,21 @@ class ConnectionKeepAlive(
+                        // TCP心跳：通过 Rust TCP 会话发送
+                        success = try {
+                            val ctx = deviceManager.rustContextInternal
+                            if (ctx != null) {
+                                val batteryLevel = BatteryUtils.getBatteryLevel(deviceManager.contextInternal)
+                                val isCharging = BatteryUtils.isCharging(deviceManager.contextInternal)
+                                val battery = if (isCharging) batteryLevel else -batteryLevel
+                                val displayName = deviceManager.localDisplayNameInternal()
+                                NativeCore.sendHeartbeatTcp(ctx, deviceManager.uuid, displayName, deviceManager.listenPort.toShort(), battery, "android")
+                            }
+                            true
+                        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)